From c4ba6a5d736b1e7b1a31982a9e91af2635e73856 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:26:51 -0400
Subject: [tools] perf/evsel: Add a debug print if perf_event_open fails

Message-id: <1408462094-14194-243-git-send-email-jolsa@redhat.com>
Patchwork-id: 88244
O-Subject: [PATCH RHEL7.1 BZ1131394 242/325] perf evsel: Add a debug print if perf_event_open fails
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit f852fd621ca19f557f2e3d05900366be7c7afb83
Author: Adrian Hunter <adrian.hunter@intel.com>
Date: Fri Nov 1 15:51:29 2013 +0200

description
===========
There is a debug print (at verbose level 2) for each call to
perf_event_open.  Add another debug print if the call fails, and print
the error number.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/util/evsel.c b/tools/perf/util/evsel.c
index 3a334f0..f0e65de 100644
--- a/tools/perf/util/evsel.c
+++ b/tools/perf/util/evsel.c
@@ -1051,6 +1051,8 @@ retry_open:
              group_fd, flags);
    if (FD(evsel, cpu, thread) < 0) {
     err = -errno;
+    pr_debug2("perf_event_open failed, error %d\n",
+       err);
     goto try_fallback;
    }
    set_rlimit = NO_CHANGE;
-- 
1.7.1